tempto-core/src/test/resources/configuration/global-configuration-tempto.yaml

```diff
@@ -2,6 +2,7 @@ value:
   default: ${not.exisiting.key:-default_value}
   both: global
   global: global
+  with.dot: 1
```

Comment:
why can't you just use:
```
with:
  dot: 1
```
?

Reply:
Having properties like:
```
"a.b.c": value1
"a.b.d": value2
"a.e.f": value3
```
which is simply 1:1 mapping, then I would get:

```
a:
  b:
    c: value1
    d: value2
  e:
    f: value3
```

Then I would need to traverse the tree to restore 1:1 mapping. Also traversing the the tree I would need to skip keys like `a.b` or `a.e` which do not have value. So it would be much more complicated than it is needed. Also notice that dots in the key name are allowed.

Reply:
But does `"a.b"` means the same as `a.b` when accessing yaml properties via path?
For example:
```
"a.b":
 c
```
does
```
"a.b".c
a.b.c
```
mean the same when accessing the value?

Reply:
That is is the problem of how tempto is addressing configuration entries in configuration. For example spring is using some escaping. See https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-Configuration-Binding#map-based-binding

> If the key contains a dot, you must use the bracket notation if it is a nested Map. For example, for the above scenario, a valid example that will use the key bar.baz is

Currently in tempto there is no way to distinguish that.

![image](https://user-images.githubusercontent.com/7569403/71512991-d7453b00-2898-11ea-9f9b-4e9db2e52038.png)

Reply:
I'm still not sure what problem you are trying to solve. What's the issue with nested trees?